src/zjs_aio.c

```diff
@@ -19,7 +19,7 @@ static struct nano_sem aio_sem;
 
 typedef struct aio_handle {
     jerry_value_t pin_obj;
-    int32_t callback_id;
+    int16_t callback_id;
```

Comment:
I see. A compiler would by default align structure fields using architecture's natural alignment, which for any 32-bit platform would be 4 bytes. Sticking 16-bit value between 32-bit and 64-bit value wouldn't save you anything. You'd need to group 2 16-bit values together to fill the slot.

You can ask compiler for a packed structure, then you save, but at the expense of the combination of the following: speed, code size, crashing due to alignment fault, or just getting invalid data.